provision.yml

```diff
@@ -7,147 +7,181 @@
     tags:
      - always
 
-
-- hosts: loadbalancer:loadbalancer-vm:php-apps:java-apps:storage:oidc:dbcluster:sysloghost:elk:lifecycle
+- hosts: base
   gather_facts: yes
   become: true
   roles:
-    - { role: common,       tags: ['common'] }
-    - { role: rsyslog,      tags: ['rsyslog'] }
-    - { role: iptables, when: iptables_enable | bool,  tags: ['iptables'] }
-
-- hosts: selfsigned_certs
-  gather_facts: no
-  become: true
-  roles:
-    - { role: selfsigned_certs, tags: ['selfsigned_certs'] } 
-
-- hosts: php-apps-vm
-  become: true
-  roles:
-    - { role: hosts,     tags: ['hosts'] }
-    - { role: httpd,     tags: ['httpd'] }
-    - { role: welcome,   tags: ['welcome'] }
-
-- hosts: loadbalancer-vm
-  gather_facts: true
-  become: true
-  roles:
-    - { role: haproxy, tags: ['lb'] }
+    - role: common
+      tags: ['core', 'base', 'common']
+    - role: rsyslog
+      tags: ['core', 'base', 'rsyslog']
+    - role: iptables
+      when:
+        - iptables_enable | bool
+      tags: ['core', 'base', 'iptables']
+    - role: selfsigned_certs
+      when:
+        - use_selfsigned_certs
+      tags: ['core', 'base', 'selfsigned_certs']
 
-- hosts: loadbalancer_ha
-  gather_facts: true
+- hosts: frontend
   become: true
   roles:
-    - { role: keepalived, tags: ['keepalived'] }
-    - { role: bind, tags: ['bind'] }
+    - role: hosts
+      tags: ['core', 'frontend', 'hosts']
+    - role: httpd
+      tags: ['core', 'frontend', 'httpd']
+    - role: welcome
+      tags: ['core', 'frontend', 'welcome']
 
 - hosts: loadbalancer
   gather_facts: true
   become: true
   roles:
-    - { role: haproxy, tags: ['lb'] }
-
-- hosts: php-apps:java-apps:oidc:lifecycle
-  gather_facts: no
-  become: true
-  roles:
-    - { role: httpd,     tags: ['httpd'] }
-
-- hosts: storage
+    - role: haproxy
+      when:
+        - inventory_hostname not in groups['loadbalancer_ha']
+      tags: ['core', 'loadbalancer', 'lb']
+    - role: keepalived
+      when:
+        - inventory_hostname in groups['loadbalancer_ha']
+      tags: ['core', 'loadbalancer', 'keepalived']
+    - role: bind
+      when:
+        - inventory_hostname in groups['loadbalancer_ha']
+      tags: ['core', 'loadbalancer', 'bind']
+
```

Comment:
Same here as for the previous plays: The role "haproxy" is not optional, but the "bind" and "keepalived" are. They are there to deploy the specific roles needed to run OpenConext over several datacenters. We might place these behind feature toggles as well.

Reply:
But the role will only be executed when the host is part of the `loadbalancer_ha` group. So if that is empty, then it won't be installed.

Reply:
On the loadbalancer_ha group all three roles need to be executed (haproxy, bind and keepalived) . In this setup, haproxy would not be installed on the production loadbalancers :)

Reply:
Applied the change.